### PR TITLE
Add Winui 2.2 framework package

### PR DESCRIPTION
**Pull request by licanhua**
Complete part of #3521. CLI templates change will be in another PR.

This is the integration with WinUI 2.2.190917002 framework package.
It’s a breaking change, and you may be impacted:
You are using an old version of WinUI like 2.1.
	You are using WinUI prelease
	Your product is not shipped from Store. You may have to install WinUI by yourself.

1. _WinUIWorkaround is added to workaround problem [WinUI issue 1512](https://github.com/microsoft/microsoft-ui-xaml/issues/1512).
2. All our name space is 'react.uwp', so I changed 'RootNamespace' to make mdmerge.exe happy. and CppWinRTNamespaceMergeDepth is added to avoid compilation error because of RootNamespace change.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3622)

### CODE REVIEW

File: vnext/ReactUWP/ReactUWP.vcxproj [hunk 1]
```diff
@@ -447,6 +448,16 @@
...
     <Error Condition="!Exists('$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.190530.8\build\native\Microsoft.Windows.CppWinRT.props')" Text="$([System.String]::Format('$(ErrorText)', '$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.190530.8\build\native\Microsoft.Windows.CppWinRT.props'))" />
     <Error Condition="!Exists('$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.190530.8\build\native\Microsoft.Windows.CppWinRT.targets')" Text="$([System.String]::Format('$(ErrorText)', '$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.190530.8\build\native\Microsoft.Windows.CppWinRT.targets'))" />
+    <Error Condition="!Exists('$(SolutionDir)packages\Microsoft.UI.Xaml.2.2.190917002\build\native\Microsoft.UI.Xaml.targets')" Text="$([System.String]::Format('$(ErrorText)', '$(SolutionDir)packages\Microsoft.UI.Xaml.2.2.190917002\build\native\Microsoft.UI.Xaml.targets'))" />
+  </Target>
+  <Target Name="_WinUIWorkaround" AfterTargets="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <!-- This is a workaround before https://github.com/microsoft/microsoft-ui-xaml/issues/1512 is resolved and the new WinUI framework package is released -->
```
Comment (chrisglein) [inline vnext/ReactUWP/ReactUWP.vcxproj]: Create a tracking issue on this repo to remove this?

Comment (licanhua) [reply vnext/ReactUWP/ReactUWP.vcxproj]: https://github.com/microsoft/react-native-windows/issues/3695

File: vnext/ReactUWP/packages.config [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="boost" version="1.68.0.0" targetFramework="native" />
+  <package id="Microsoft.UI.Xaml" version="2.2.190917002" targetFramework="native" />
```
Comment (stecrain) [inline vnext/ReactUWP/packages.config]: Seems like this is a nuget package that contains a framework package?

**Comment by stecrain** [comment]
Are there any changes to the cli template needed as part of this?

File: vnext/ReactUWP/packages.config (hunk 2 above)
Comment (licanhua) [reply vnext/ReactUWP/packages.config]: yes. It's framework package

---
In reply to: [349762583](https://github.com/microsoft/react-native-windows/pull/3622#discussion_r349762583) [](ancestors = 349762583)

**Comment by licanhua** [comment]
> Are there any changes to the cli template needed as part of this?
> Since you are advertising it as a breaking change I assume the cli template would need to be updated 🤔

Good call. I reword the description. and it will be in the next PR

File: vnext/ReactUWP/Views/module.g.cpp [hunk 3]
```diff
@@ -1,19 +1,20 @@
-﻿// WARNING: Please don't edit this file. It was generated by C++/WinRT v1.0.190111.3
+// WARNING: Please don't edit this file. It was generated by C++/WinRT v1.0.190111.3
```
Comment (dotnetjunky) [inline vnext/ReactUWP/Views/module.g.cpp]: This is not true. We should just delete this line.

Comment (licanhua) [reply vnext/ReactUWP/Views/module.g.cpp]:
```suggestion
```

**Pull request merged**